src/cpp/server/ServerSessionManager.cpp

```diff
@@ -382,12 +382,33 @@ Error launchSession(const r_util::SessionContext& context,
                     PidType* pPid)
 {
    // launch the session
+   // we use a modified configured home directory to provide a realiable temp dir that can be written to
+   // as the server (service) user most likely does not have a home directory configured
    std::string username = context.username;
    std::string rsessionPath = server::options().rsessionPath();
    std::string runAsUser = core::system::realUserIsRoot() ? username : "";
    core::system::ProcessConfig config = sessionProcessConfig(context,
                                                              extraArgs);
 
+   FilePath tmpDir;
+   Error error = FilePath::tempFilePath(tmpDir);
+   if (error)
+   {
+      LOG_ERROR(error);
+   }
+   else
+   {
+      error = tmpDir.ensureDirectory();
+      if (error)
+      {
+         LOG_ERROR(error);
+      }
+      else
+      {
+         config.environment.push_back(std::make_pair("XDG_CONFIG_HOME", tmpDir.getAbsolutePath()));
```

Comment:
Should this use setenv (so that it overrides any existing `XDG_CONFIG_HOME`)?

Reply:
Good catch, will fix that.